ouroboros-network/src/Ouroboros/Network/Mux.hs

```diff
@@ -100,11 +100,13 @@ muxStart udesc bearer = do
                                 ModeResponder
                                 w_r cnt
 
-        return $ map (>> mpsJobExit cnt) $ catMaybes
-          [ mpdInitiator mpd <*> Just initiatorChannel
-                
-          , mpdResponder mpd <*> Just responderChannel
-          ]
+        return $ case app of
+          MuxInitiatorApplication initiator -> [ initiator mpdId initiatorChannel >> mpsJobExit cnt ]
+          MuxResponderApplication responder -> [ responder mpdId responderChannel >> mpsJobExit cnt ]
+          MuxInitiatorAndResponderApplication initiator responder
+                                            -> [ initiator mpdId initiatorChannel >> mpsJobExit cnt
+                                               , responder mpdId responderChannel >> mpsJobExit cnt
```

Comment:
Initiator channel will write to the underlying bearer `MuxSDU`s with `msMode == ModeInitiator` and it will read `MuxSDU`s which come from network with `msMode == ModeResponder` (i.e. where send by the responder).

Responder channel will work in a dual way (write `ModeResponder` sdus and read `ModeInitiator` sdus).